summingbird-core/src/main/scala/com/twitter/summingbird/planner/Node.scala

```diff
@@ -78,12 +78,45 @@ case class SourceNode[P <: Platform[P]](override val members: List[Producer[P, _
   override def shortName(sanitize: String => String) = NodeIdentifier("Source" + collapseNamedNodes(sanitize))
 }
 
-case class Dag[P <: Platform[P]](originalTail: TailProducer[P, _], producerToPriorityNames: Map[Producer[P, Any], List[String]], tail: TailProducer[P, _], producerToNode: Map[Producer[P, _], Node[P]],
-    nodes: List[Node[P]],
-    nodeToName: Map[Node[P], String] = Map[Node[P], String](),
-    nameToNode: Map[String, Node[P]] = Map[String, Node[P]](),
-    dependenciesOfM: Map[Node[P], List[Node[P]]] = Map[Node[P], List[Node[P]]](),
-    dependantsOfM: Map[Node[P], List[Node[P]]] = Map[Node[P], List[Node[P]]]()) {
+/**
+ * Summingbird converts a Directed Acyclic Graph(DAG) of Producers into a DAG of Nodes.
+ * Multiple producers can get mapped into a single Node as a result of optimization. This
+ * class keeps track of these mappings and bunch of other information. The Dag in question
+ * here is pivoted on a single tail producer, in other words it is the view from a single
+ * TailProducer, other TailProducers may have their own view.
+ *
+ * Dag objects aren't typically constructed directly, but rather through the companion
+ * object helpers, which also generate unique names for each Node.
+ *
+ * @param originalTail                The TailProducer before stripping out named nodes.
+ * @param producerToPriorityNames     Map every producer to list of names that apply to it.
+ *                                    These are the names provided by named nodes.
+ *                                    e.g. .name("mynode"). Names flow backwards, i.e from
+ *                                    source to tail.
```

Comment:
Can you clarify the "Names flow backwards, from source to tail" - Actually I think that a Source inherits the name from the Tail